### ISSUE TITLE
Beaglebone Black Wireless fails to boot after custom device tree blob change (+ connection issues with the other Beaglebone models → scroll down)

### ISSUE THREAD

**Issue by thmsmlt** [ctag-fh-kiel/ctag-face-2-4#2]
Let's start this issue where the other one ended: I'm trying to get the AD1938 working with a BeagleBone Black Wireless. I flashed the BBBW with a recent official BBBW image and tried to use the custom device tree blob presented in this repository. However after changing uEnv.txt my BBBW refuses to boot.

When adding/substituting (I tried both) this line: 
`dtb=am335x-bonegreen-ctag-face.dtb`
The BBB won't boot and gets stuck with the usr0 led blinking

I get the same thing when using the given quick start image by the way.

When adding/substituting (I tried both) this line as well as the above:
`uname_r=4.1.18+`
The BBB won't boot and gets stuck with all usr leds on forever.

I first tried to do these adaptations with the OS on the SD-card. Then I also tried to copy the OS to the internal memory first and then doing the adaptations, but (as expected) no change there, so that's not the issue.

Any ideas on what's going wrong?
How long should I wait with the led blinking until giving up, my last attempt ran for an hour and a half... What's the booting time of a working solution?

**Comment by henrix** [comment]
I think this is caused from the device tree.
I will create a new device tree for the BeagleBone Black Wireless the next days, but unfortunately I don't have a BBB Wireless for testing.

I will update this thread then and hope it works.

Moreover I will create a stable 4.4 branch and move my active development for the daisy chained driver to a dedicated branch.

**Comment by thmsmlt** [comment]
That's very good news.

I'll be using a BeagleBone Black (non Wireless) in the meantime - or do you recommend the Green?

(To clarify: I'm using an AD1938 as a part of my master's thesis and I'm trying to find a good platform to demo my hardware ;) )

**Comment by henrix** [comment]
You could use the Black or Green.
Both are working fine.

Glad to hear that you use our platform;)

**Comment by thmsmlt** [comment]
Hi, I've tried both the Black and Green but to no avail. 

The quick start image still won't boot and when using a beagleboard.org image, the black behaves in the same way as the black wireless and the green won't even boot with the unmodified image. Surely I must be doing something wrong. Could you give a more detailed description of the steps that need to be taken? I hope I'm making a small mistake here ;)

Some more questions:
- How do you interface with the boards: USB, Ethernet, HDMI & Keyboard (in the case of the BBB)...?
- When used with USB, which computer do you use? (I've tried on a windows laptop and a macbook)
- When used with ethernet, are there special connection steps to be taken?
- When used with HDMI, does your BBB give a terminal window when connected to a screen?
- How long does it take for the quickstart image to boot?

Update: when using the quickstart image on SD in a BBB, connected via Ethernet, my windows computer's network troubleshooter says that "Ethernet does not have a valid IP configuration"

**Comment by henrix** [comment]
Hi,

I have created a [new image](https://drive.google.com/file/d/0B2goLqs_HZ3QRXdhWno2cUsta0U/view?usp=sharing) last week. You don't have to change anything, to use it with the AD1938 codec.

On the official images, you have to:
1. Edit the dtb= line in the uEnv.txt file and change it to am335x-boneblack-ctag-face.dtb or am335x-bonegreen-ctag-face.dtb
2. Compile the device tree overlay (either the overlays for software SPI or hardware SPI) and load it via the Bone Capemanager

> How do you interface with the boards: USB, Ethernet, HDMI & Keyboard (in the case of the BBB)...?

I personally use the virtual Ethernet via USB and ssh due to flexibility.

> When used with USB, which computer do you use? (I've tried on a windows laptop and a macbook)

I'm using a Linux distribution. If you want to use Windows or Mac OS, you have to [install the drivers](http://beagleboard.org/getting-started) for virtual Ethernet.

> When used with ethernet, are there special connection steps to be taken?

The Ethernet port of the BB is configured to use an automatically assigned IP. So you need a DHCP server.

> When used with HDMI, does your BBB give a terminal window when connected to a screen?

I don't use HDMI. In our device tree, HDMI is disabled, as well.

> How long does it take for the quickstart image to boot?

It should be around 20 seconds (depends if you are using a GUI or running the BB headless). Moreover there are some performance optimizations available.

> Update: when using the quickstart image on SD in a BBB, connected via Ethernet, my windows computer's network troubleshooter says that "Ethernet does not have a valid IP configuration"

You don't have a DHCP server running. You can try [this one](http://www.dhcpserver.de/cms/) on Windows.

**Comment by thmsmlt** [comment]
Thanks for the answers! 

I've been trying and searching around and here's where I am now: 
I did install all drivers but the DHCP seems to do things beyond my comprehension...

- On windows, I installed a DHCP server and I do get a connection, but when I try to ssh to the given IP address with putty, I get a connection timed out error, I also tried the default IP adresses (192.168.6.1, 192.168.7.1, 192.168.7.2, 192.168.8.1) , but to no good
- On linux (Ubuntu 14.04) , two Beaglebone virtual ethernet connections are active, but the same problem, I can't find the right IP address, tried nmap for a ton of addresses, but no luck there...
- On mac, I also get two connections, but with the warning "self assigned IP", no SSH command seems to get the connection running

Which brings me to the next question:

Given that the Beaglebone gets it's IP address dynamically with DHCP, where can I find the right IP address for SSH? Or what's the name of the BBB for SSH (e.g. ssh debian@beaglebone doesn't work)?

Many thanks!

**Comment by henrix** [comment]
The Ethernet port of the BeagleBone is configured to use a DHCP server by default.
The virtual Ethernet port via USB of the BeagleBone has a static IP (192.168.7.2) and should assign 192.168.7.1 to the virtual Ethernet port of your host PC.
So, when using the virtual Ethernet via USB, you should be able to ssh into your BeagleBone with `ssh debian@192.168.7.2` and the default password "temppwd".

**Comment by thmsmlt** [comment]
Unfortunately, when connected with USB, all platforms give a connection timeout for `ssh debian@192.168.7.2`. But I cracked it on windows. After setting up my DHCP server, I went into the networks and sharings center and changed the IP address for the windows computer for communicating via ipv4 to 192.168.7.1.

 https://www.youtube.com/watch?v=D-NEPiZDSx8 → it's shown in the middle of the video

Thanks for all the help :) let's hope I get everything running now :)

Update: works on mac os too: went in to system preferences: network, changed the "Configure ipv4" for Beaglebone Black to “manual” and changed the IP address to 192.168.7.1, pressed Apply et voila :)
→ When terminal gives the error "Remote Host Identification Has Changed" run `ssh-keygen -R 192.168.7.2`
